packages/core-flows/src/product/steps/batch-products.ts

```diff
@@ -0,0 +1,75 @@
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+import { createProductsWorkflow } from "../workflows/create-products"
+import { updateProductsWorkflow } from "../workflows/update-products"
+import { deleteProductsWorkflow } from "../workflows/delete-products"
+import { PricingTypes, ProductTypes } from "@medusajs/types"
+
+type WorkflowInput = {
+  create: (Omit<ProductTypes.CreateProductDTO, "variants"> & {
+    sales_channels?: { id: string }[]
+    variants?: (ProductTypes.CreateProductVariantDTO & {
+      prices?: PricingTypes.CreateMoneyAmountDTO[]
+    })[]
+  })[]
+  update: (ProductTypes.UpsertProductDTO & {
+    sales_channels?: { id: string }[]
+  })[]
+  delete: string[]
+}
+
+export const batchProductsStepId = "batch-products"
+export const batchProductsStep = createStep(
+  batchProductsStepId,
+  async (data: WorkflowInput, { container }) => {
+    const { transaction: createTransaction, result: created } =
+      await createProductsWorkflow(container).run({
+        input: { products: data.create },
+      })
+    const { transaction: updateTransaction, result: updated } =
+      await updateProductsWorkflow(container).run({
+        input: { products: data.update },
+      })
+    const { transaction: deleteTransaction } = await deleteProductsWorkflow(
+      container
+    ).run({
+      input: { ids: data.delete },
+    })
+
+    return new StepResponse(
+      {
+        created,
+        updated,
+        deleted: {
+          ids: data.delete,
+          object: "product",
+          deleted: true,
+        },
+      },
+      { createTransaction, updateTransaction, deleteTransaction }
+    )
+  },
+
+  async (flow, { container }) => {
+    if (!flow) {
+      return
+    }
+
+    if (flow.createTransaction) {
```

Comment:
question: Do we know if this is the correct approach to handle such a scenario? If so, I think we should add this in our conventions doc.

Reply:
@olivermrbl as far as Adrien and Carlos were concerned, then yes this is how it works. I will add more testing once this is merged just to see if everything works as expected, but I wanted to at least have these methods available for the admin to be able to do pricing updates.

I'll also add it to the conventions, I'll add a ticket for myself.

Reply:
Cool, sounds good